api/lib/jwt_service.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'jwt'
+
+# JWTService
+module TokenAuthentication
+  # encoding data into a jwt token
+
+  def self.encode(payload, exp = 24.hours.from_now)
+    payload[:exp] = exp.to_i
+    JWT.encode(payload, Rails.application.secrets.secret_key_base)
```

Comment:
Instead of having a secret key for the encoding and decoding, use an environment variable instead.